hail/src/main/scala/is/hail/asm4s/CodeBuilder.scala

```diff
@@ -33,13 +33,9 @@ trait CodeBuilderLike {
 
   def result(): Code[Unit]
 
-  val localBuilder: SettableBuilder = new SettableBuilder {
-    def newSettable[T](name: String)(implicit tti: TypeInfo[T]): Settable[T] = mb.newLocal[T](name)
-  }
+  def localBuilder: SettableBuilder = mb.localBuilder
```

Comment:
Just a drive-by remark: I kind of regret choosing the name `localBuilder`.  I think it should just be `newLocal _`, where newLocal matches the exsiting local creation functions we have.  Then SettableBuilder[T] is just a type alias (if it needs exists at all) for `(String) => (TypeInfo[T]) => Settable[T]`.

Think about it.

Reply:
I'm not sure what the type signature of the lambda would be in order to write something like
```scala
// Somewhere
type SettableBuilder = (String) => TypeInfo[???] => Settable[???] // we need the callers of this to choose ???
// EmitFunctionBuilder.scala
def newPSettable(_pt: PType, mkSettable: SettableBuilder, baseName: String): PSettable
// PCanonicalArray.scala
Object PCanonicalIndexableSettable {
  def apply(mkSettable: SettableBuilder, pt: PContainer, name: String): PCanonicalIndexableSettable = {
    new PCanonicalIndexableSettable(pt,
      mkSettable(s"${ name }_a")(LongInfo),
      mkSettable(s"${ name }_length")(IntInfo),
      mkSettable(s"${ name }_elems_addr")(LongInfo))
  }
}
```
I'm more than willing to do this. It feels like the right thing to do, but I'm not sure how to actually write the scala code such that it type checks.

Reply:
I'm not sure this can work. The type that scala infers for `newLocal _` is `(string) => LocalRef[Nothing]`.

Reply:
I still haven't figured this out, so I'm continuing on with `SettableBuilder`